composer/trainer/trainer.py

```diff
@@ -807,6 +806,11 @@ def eval(self, is_batch: bool):
 
             metrics = self._get_metrics_as_collection(is_train=False)
 
+            if isinstance(self.state.eval_dataloader.sampler, torch.utils.data.DistributedSampler):
+                # using the batch as the epoch, as the evaluator may be running every batch
+                # instead of every epoch
+                self.state.eval_dataloader.sampler.set_epoch(int(self.state.timer.batch))
+
```

Comment:
What is this section of code meant for? The eval dataloader's sampler should probably not matter. And setting the epoch to the current batch seems a bit strange?

Reply:
Discussed offline; updated comment.